### DISCUSSION TITLE
How do I close the database connection?

### DISCUSSION THREAD

**Discussion by efrain2007** [SeaQL/sea-orm#608] [Q&A] [answered]
How to close the connection after using it?
what I want to achieve is to connect to multiple databases at the same time

**Comment by billy1624** [comment, marked as answer]
Hey @efrain2007, sorry for the delay. SQLx have [`close`](https://docs.rs/sqlx/0.5.11/sqlx/struct.Pool.html#method.close) method but SeaORM does not expose this method on public API. So, I would suggest to create a new connection inside a scope then exit the scope and the connection pool will be dropped correspondingly.

**Reply by hunjixin** [reply]
I have a question, for sqlite, after I close the program, the journal file is still very large, even much larger than db file. I'm not quite sure if this is reasonable

```
staff   200K Mar 23 12:13 gpuproxy.db
staff    32K Mar 23 12:05 gpuproxy.db-shm
staff   3.9M Mar 23 12:14 gpuproxy.db-wal
```

**Reply by billy1624** [reply]
Hey @hunjixin, it's normal. And you can safely delete the temporary files
- https://www.sqlite.org/tempfiles.html

**Reply by efrain2007** [reply]
@billy1624 
Some example of how to implement

**Reply by darkskygit** [reply]
It should be noted that for sqlite with `WAL` mode, wal files will be safely deleted after all connections are closed. If you exit the program without closing all connections, it may damage the database, so we need to clearly drop all sqlite connections.
Can the current version of sea-orm correctly close all sqlite connection when dropping DatabaseConnection?

**Comment by ohaddahan** [comment]
I was creating new `connections` and expecting them to be closed automatically or on explicit `drop` or via `max_lifetime` or `idle_timeout`.
Seems like non worked and I was reaching my DB `connections` limit and everything was getting blocked.
I think there might be some issue.

I resolved it by using https://github.com/matklad/once_cell : 
```rust
static DB: OnceCell<Box<DatabaseConnection>> = OnceCell::new();
```

And passing the same DB connection.

**Reply by billy1624** [reply]
Hey @ohaddahan, is there any reason why you create multiple `DatabaseConnection`?
`DatabaseConnection` is a connection pool and you should reuse it instead of creating a new `DatabaseConnection` every time you want to execute a query.

**Reply by ohaddahan** [reply]
Yeah, that was what I wanted to do from the start, but didn't got it to compile until I used `Box`.
That's how I use it now.

Regardless, I think this might indicate some issue, since we shouldn't leak `connections`.

**Reply by billy1624** [reply]
Alright! Thanks for the sharing!!
